### PR TITLE
Display disk usage bar chart for multiple disks

### PR DESCRIPTION
**Pull request by rszwajko**
Changes:
1. create chart labels from valid 'datum' object. It's now passed as a
    nested object.
2. use constant offsetX=0 for bar chart. Before it was calculated based
   on longest  disk 'path' which caused the chart to be moved out of
   the view for long paths.
3. add disk 'path' to tooltip. It's too long to be displayed on X axis.
4. display used or available disk inside tooltip depending on mouse
   pointer location. Before, if usage was 100% the tooltip was missing.

### CODE REVIEW

**Comment by rszwajko** [comment]
### Problem re-created
disku_usage_wrong_tooltip
### Effect of long path (chart is out of view)
disk_usage_out_of_view
### Fixed
disk_usage_fixed

File: src/components/VmDetails/cards/UtilizationCard/UtilizationCharts/BarChart.js [hunk 1]
```diff
@@ -41,13 +41,13 @@ const BarChart = ({ data, additionalLabel, thresholdWarning, thresholdError, id,
             }}
             labelComponent={<CustomLabel offsetX={offsetX} label={additionalLabel} />}
             data={data}
-            {...rest}
+            labels={({ datum }) => `${msg.utilizationCardLegendUsed()} ${datum.y}%  (${datum.x})`}
```
Comment (sjd78) [inline src/components/VmDetails/cards/UtilizationCard/UtilizationCharts/BarChart.js:44]: So just FYI, the percent symbol location and spacing can change per locale, so in ui-extensions we have a message like this:
https://github.com/oVirt/ovirt-engine-ui-extensions/blob/8a0269401791f58667d93d43d3a1728db08f9457/src/intl/messages.js#L89-L93

I know you just reused the label, but that formatting should be translatable.

**Review by sjd78 — COMMENTED** [review]
The issue is fixed.

2 comments:
  1. What about the different localization string for the label text?
  2. How about moving the tooltip location?  To the right looks a bit strange with a long filesystem mount point.  Above or below may be a better choice.
  
screenshot-localhost_3000-2021 12 03-10_23_32

**Comment by rszwajko** [comment]
@sjd78 I was able to put the tooltip at the top of the bar. However it's still placed at the end of the bar. I've found no way to change this position (other the hard-coded offsets)

image
image
image

**Review by sjd78 — APPROVED** [review]
LGTM!

IMHO, the top orientation of the tooltip does look better.  @sgratch , any strong opinion on the tooltip placement?  I don't think it is worth trying to tweak the location any further than what Radek already did.

**Review by sgratch — DISMISSED** [review]
Generally looks good and solves the problem. Few comments:

> @sjd78  IMHO, the top orientation of the tooltip does look better. @sgratch , any strong opinion on the tooltip placement? I don't think it is worth trying to tweak the location any further than what Radek already did.

The top tooltip orientation looks good to me as well, let's leave it like that.

**Note**: There is a problem with disk's thresholds colors - they are incorrect and colored in red for 23% usage.

It seems that the following lines still use old percentage values (e.g. 140%) instead of the correct ones:
https://github.com/oVirt/ovirt-web-ui/blob/5d059762b8f55ef1a7a9db6b0e113286ba9aaed6/src/components/VmDetails/cards/UtilizationCard/UtilizationCharts/BarChart.js#L35-L39

File: src/intl/messages.js [hunk 2]
```diff
@@ -723,8 +723,16 @@ export const messages: { [messageId: string]: MessageType } = {
   },
   utilizationCardLegendAvailable: 'Available',
   utilizationCardLegendAvailableP: '% Available',
+  utilizationCardLegendAvailableWithDetails: {
+    message: 'Available {value}% ({diskPath})',
+    description: 'Details about free disk space. Example: Available 75% (/boot)',
+  },
   utilizationCardLegendUsed: 'Used',
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/intl/messages.js:730]: The trade off is either:
1. Using existing messages that are already translated, or
2. Using new messages that allow the order of things to be "translated" as well.

The "right" thing is option 2, but the least disruptive to translations is option 1.

@sgratch - I'm 100% ok with leaving the new messages there and having an untranslated key in zanata.  I expect there will be other untranslated things in 4.5, I just don't know if there will be a translation cycle like normal.  With a translation cycle, option 2 is best and no change is needed.

**Comment by sjd78** [comment]
>
> **Note**: There is a problem with disk's thresholds colors - they are incorrect and colored in red for 23% usage.
> 
> It seems that the following lines still use old percentage values (e.g. 140%) instead of the correct ones:
> 
> https://github.com/oVirt/ovirt-web-ui/blob/5d059762b8f55ef1a7a9db6b0e113286ba9aaed6/src/components/VmDetails/cards/UtilizationCard/UtilizationCharts/BarChart.js#L35-L39

This is a fix outside the scope of this PR.  I opened #1556 to cover the bar chart colors.

File: src/intl/messages.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/intl/messages.js:730]: @sgratch  @sjd78 
IMHO creating a new message is the right thing here. The tooltip should remain readable even if the string will remain untranslated (until next translation cycle).

**Review by sgratch — COMMENTED** [review]
Regarding changing the tooltip text:
There unfortunately won't be another translation cycle for 4.5, at least they are not planning such for now. 

So fixing messages is really depend on necessarily and how important this message is (how frequent it appears etc).
In our case, this is not a hidden field text or a specific notification. The fixed tooltip is part of VM dashboard card and includes an important info of used percentage and disk's file system. it's hard to read the disk utilizatin info without it. I really don't think we want to break that translation if not necessary. 

But this can be easily fix. For new `msg.utilizationCardLegendUsedWithDetails`: we can take the "Used" from old translations, push it to the new translations manually, leaving the diskPath as well since anyway it's not translated. Then push the fixed new translation to Zanata:
https://github.com/oVirt/ovirt-web-ui/blob/5d059762b8f55ef1a7a9db6b0e113286ba9aaed6/src/intl/messages.js#L732-L734

Regarding the 2nd tooltip new message `msg.utilizationCardLegendAvailableWithDetails`: we can leave it as is untranslated since it's totally new and it's better to add this than not. We can also translate it manually based on old translations   `utilizationCardLegendAvailable: 'Available'`, but here it's less necessary.
https://github.com/oVirt/ovirt-web-ui/blob/5d059762b8f55ef1a7a9db6b0e113286ba9aaed6/src/intl/messages.js#L726-L728

**Comment by rszwajko** [comment]
> But this can be easily fix

Sure. We can do it together with other tasks that require adjusting translations i.e. https://bugzilla.redhat.com/show_bug.cgi?id=2024202

**Comment by sgratch** [comment]
> > But this can be easily fix
> 
> Sure. We can do it together with other tasks that require adjusting translations i.e. https://bugzilla.redhat.com/show_bug.cgi?id=2024202

So once the translation files are ready, we can merged this and only then push to Zanata the new translated strings.

**Comment by ovirt-infra** [comment]
All tests passed

**Review by sjd78 — APPROVED** [review]
Translation adjustments look good.

We can merge and then handle the zanata sync up later (with the other PRs that will need similar work).

**Comment by isaranova** [comment]
LGTM

**Pull request merged**